osc/build.py

```diff
@@ -1351,8 +1354,8 @@ def __str__(self):
                 print("Error: cannot get hdrmd5 for %s" % i.fullfilename)
                 sys.exit(1)
             if hdrmd5 != i.hdrmd5:
-                print("Error: hdrmd5 mismatch for %s: %s != %s" % (i.fullfilename, hdrmd5, i.hdrmd5))
-                sys.exit(1)
+                print("WARNING: OBS BUG hdrmd5 mismatch for %s: %s != %s" % (i.fullfilename, hdrmd5, i.hdrmd5))
```

Comment:
Is it always an OBS bug?
What if a package signature changes?
What if it gets rebuild under the same NEVRA and re-published to the same repo?

Reply:
Yes, it's always an obs bug. OBS builds multiple no-arch packages but then only md5's one of them. So when you get "one of the other" no arch packages, it's md5 will never match even though it's valid and on a mirror. 

Then the md5 should need to also change and it's gpg sig would change.

And again, then it should be rehashed and resigned if it's re-published, and should have a version build id change to match. Else mirrors of OBS don't know they need to update! 

As mentioned, this md5 seems to serve no purpose.

Reply:
just for correctness, it is not an OBS bug, it is the way the distribution publishing is designed to work in product-builder. It is just not resolvable by OBS.

The md5 sums are handled as lookup identifier by rpm, they are by no mean a security implementation.

Reply:
an alternative would be an additional switch like   --disable-hdrmd5-rpm-checking or alike to disable the check. You won't have the same setup as on the server side, but in most cases it does most likely not matter. (Only when noarch packages actually differ on the architectures, which would be then an undiscovered packaging bug).

Reply:
So then what is the check doing? It doesn't sound like it actually serve a purpose then. 

It's certainly a case where it means "anyone outside the EU" get's a bad user experience because of this while the check remains.

Reply:
> just for correctness, it is not an OBS bug, it is the way the distribution publishing is designed to work in product-builder. It is just not resolvable by OBS.

If osc always gets the wrong md5 or download location and re-downloads every time although a perfectly fine package would be available in the cache this *IS* a bug.

#1230